### PR TITLE
Rarepokemon extraction changed with respect to common schema

### PR DESCRIPTION
**Pull request by vivek-sethia**
Commit id : 5bca91ae93844c32121cd55d45bc4bf217a43074

### CODE REVIEW

File: app/stores/pokemon.js [hunk 1]
```diff
@@ -59,5 +59,18 @@ module.exports = {
...
             }
         });
+    },
+
+    /*
+     * searching the pokemon details by name
+     */
+    getByName: function (name, callback) {
+        this.get({'name': name}, function (success, message) {
+            if (success == 1) {// on successfully finding previous data
```
Comment (swathi-ssunder) [inline app/stores/pokemon.js]: Use === instead of ==

File: app/controllers/filler/rarePokemon.js [hunk 2]
```diff
@@ -90,12 +87,23 @@ module.exports = {
...
 
             async.forEach(rarePokemonRecords,function (rarePokemon, callback) {
                 length++;
-                addRarePokemon(rarePokemon, function(success){
-                    callback();
+                pokemonName = (rarePokemon.name);
+                // extracting pokemon id from the names
+                pokemonStore.getByName(pokemonName, function (success, data) {
+                    if (success == 1) {
+                        rarePokemon['id'] = data[0]['pokemonID'];
```
Comment (swathi-ssunder) [inline app/controllers/filler/rarePokemon.js]: Use === instead of ==

File: app/controllers/filler/rarePokemon.js [hunk 3]
```diff
@@ -1,107 +1,65 @@
...
+ * inserting the read data into MongoDB
+ */
+    insert : function (rarePokemons) {
+        let length = 0,
+            pokemonName;
+
+        async.forEach(rarePokemons, function(rarePokemon){
+            length++;
+            pokemonName = (rarePokemons[0].name);
+            // extracting pokemon id from the names
```
Comment (swathi-ssunder) [inline app/controllers/filler/rarePokemon.js:52]: Seems like this should be rarePokemon.name. 
rarePokemons[0].name will always consider the first object, which is not the expected behavior if you have a loop.

**Comment by vivek-sethia** [comment]
1. @swathi-ssunder  Suggested changes made
2. @samitsv @jonas-he @fabe85 @swathi-ssunder  - Now there is no file creation when the listen script is executed, instead it directly stores the data to the MongoDB as suggested by @sacdallago

**Comment by sacdallago** [comment]
So much less code :heart_eyes:

**Pull request merged**